frontend/packages/akr/src/redux/sagas/clerkTranslator.ts

```diff
@@ -32,6 +32,20 @@ function* loadClerkTranslatorsSaga() {
   }
 }
 
+export function updateClerkTranslators(
+  translators: Array<ClerkTranslator>,
+  translator: ClerkTranslator
+) {
+  const updatedTranslators = [...translators];
+  const translatorIdx = updatedTranslators.findIndex(
+    (t: ClerkTranslator) => t.id === translator.id
+  );
+
+  updatedTranslators.splice(translatorIdx, 1, translator);
+
+  return updatedTranslators;
+}
+
```

Comment:
Mun mielestä tämä funktio ei oikein kuulu näiden sagojen yhteyteen. Kun tätä tunnutaan aina käyttävän tyyliin `const updated = updateClerkTranslators(...); yield updateClerkTranslatorsState(updated);`, niin ehkä tämä koko logiikka olisi helpoin tehdä suoraan siellä reducerin puolella. Silloin nuo tätä funktiota kutsuvat sagatkin hieman yksinkertaistuisivat.

Reply:
Tämä kyllä on vähän raskas operaatio hoitaa reducerin puolella, vaikka se on synkroninen. Sagan puolella pitääminen voi olla parempi vaihtoehto.

Reply:
Tosiaan Reduxin suositukset ovat muuttuneet näköjään 2022: https://redux.js.org/style-guide/#put-as-much-logic-as-possible-in-reducers . Tämä on ok varmaan.